File: DoAnCoSo Ver 14.0/DoAnCoSo/Models/ServiceDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DoAnCoSo.Models
{
    public class ServiceDetail
    {
        [Key]
        public int ServiceDetailId { get; set; }

        [ForeignKey("Service")]
        public int ServiceId { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; } // Ví dụ: "Tắm gội cơ bản", "Cạo lông"

        [Required]
        [Range(0, double.MaxValue)]
        public decimal Price { get; set; }

        [Range(0, double.MaxValue)]
        public decimal? SalePrice { get; set; }

        // Quan hệ
        public Service Service { get; set; }
    }
}

File: AppSpaHomestayChoThuCung/lib/model/service/service_detail.dart
class ServiceDetailModel {
  final int serviceDetailId;
  final int serviceId;
  final String name;
  final double price;
  final double? salePrice;

  ServiceDetailModel({
    required this.serviceDetailId,
    required this.serviceId,
    required this.name,
    required this.price,
    this.salePrice,
  });

  factory ServiceDetailModel.fromJson(Map<String, dynamic> json) {
    return ServiceDetailModel(
      serviceDetailId: json['serviceDetailId'],
      serviceId: json['serviceId'],
      name: json['name'],
      price: (json['price'] ?? 0).toDouble(),
      salePrice: json['salePrice']?.toDouble(),
    );
  }
}